themes/gatsby-theme-blog/src/theme/styles.js

```diff
@@ -0,0 +1,73 @@
+export default {
+  root: {
+    fontFamily: `body`,
+  },
+  Container: {
```

Comment:
Some questions I can see people asking:

* What is `root`?
* Does this create a new `Container` component or apply to any `Container` component?
* This is shadowable on the theme-level, so how does it combine when multiple themes use different styles.js'

I think this styles API is going to confuse people since it basically adds API surface area that does the same thing as component shadowing while reducing the ability to discover provenance of the styles and thus deviates from typical React approaches to adding more props, etc.

Also interesting will be how we merge styles like this across themes.

```
div: {
    "&.gatsby-highlight": {
    }
}
```

Reply:
- The `styles.root` styles are for setting top-level styles and the component is only rendered with `<Styled.root>` to give you control over where these styles apply. For Typography.js styles, this is used in place of styling the `html` and `body` elements. For this PR, it's added here: https://github.com/gatsbyjs/gatsby/pull/14377/files#diff-c228aeead46e1216bd9de630c447985bR67
- `styles.Container` will apply to theme-ui's `Container` component if it's used. For this theme, it has a narrower max-width than the default styles. I'm not 100% sold on this approach, and an alternative I've thought about is defining values like that in e.g. `theme.sizes.container` instead of coupling them to a specific component. There will probably also be situations where things like header height, sidebar width, etc. come into play in other themes
- The `theme` object from multiple themes (currently) will be [deeply merged](https://github.com/system-ui/theme-ui/blob/master/theme-ui/src/core.js#L23-L27), with the inner most ThemeProvider taking precedence. The `styles` object is a shortcut for defining styled components that are consumed in MDX. Part of the idea with system-ui/theme-ui#30 is to handle what's being done here with the `wrapRootElement` API in a Gatsby plugin so that we can change how that works without causing breaking changes in themes like this one and `gatsby-theme-notes`. If you've installed both the blog and notes theme, I think the idea would be that your site ends up with one merged `theme` object that both themes consume, making the `Container` component the same `max-width` for the entire site. I'm open to exploring alternatives to this (maybe theme-ui components are provided as a Gatsby theme that can be shadowed?), and can open an issue on the `theme-ui` repo to discuss different approaches.

As for the `div > .gatsby-highlight` style, I suspect we might want to look into updating the output HTML from `gatsby-remark-prismjs` to better fit in with how MDX works. Curious if you have any thoughts there, @ChristopherBiscardi.

Reply:
@jxnblk good to merge this and continue pondering this outside of this PR?

Reply:
Yeah definitely, mostly wanted to provide some context here, but not blocking